app/models/products/user_note_mode.rb

```diff
@@ -0,0 +1,49 @@
+module Products
+
+  class UserNoteMode
```

Comment:
It sort of came out of an experimentation with "whole value objects" (if you have a ruby tapas subscription: https://www.rubytapas.com/2016/04/14/episode-401-whole-value/), and I ended up liking this as it encapsulated the concept of a "mode" pretty well, and is a lot more versatile than . Otherwise a lot of this code would be tacked on inside of Product.

It's not really doing any validation here, just telling you what the valid modes are (I could maybe come up with with better names for the two class methods). And realy `valid_values` is kind of a private method: you should use `values` for getting a list of valid `UserNoteMode` instead of strings.

The only real presenter-y part of this is `to_label`, which is a way of getting a localized version of the value. To me, it's similar to `human_attribute_name` or `model_name`. Calling it `to_label` is to give it an API that simple_form will use before it tries `to_s`.

As far as why it's in models and not lib is to me this is application/domain logic, not really library code (I would consider library code to be things that could conceivably be pulled into a separate gem, used in an other project and still make sense). To be fair, we do have some things in `lib` that probably don't belong there (e.g. `AffiliateAccount` would probably live better in `app/models/concerns`).

Does that all make sense?